group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -0,0 +1,1040 @@
+                                    log.info("Finished loading of metadata from {} with epoch {}.",
+                                        tp, partitionEpoch);
+                                } catch (Throwable ex) {
+                                    log.error("Failed to load metadata from {} with epoch {} due to {}.",
+                                        tp, partitionEpoch, ex.toString());
+                                    ctx.transitionTo(CoordinatorState.FAILED);
+                                }
+                            });
+                        });
+                        break;
+
+                    case LOADING:
+                        log.info("The coordinator {} is already loading metadata.", tp);
+                        break;
+
+                    case ACTIVE:
+                        log.info("The coordinator {} is already active.", tp);
+                        break;
+
+                    default:
+                        log.error("Cannot load coordinator {} in state {}.", tp, context.state);
+                }
+            } else {
+                log.info("Ignoring loading metadata from {} since current epoch {} is larger than or equals to {}.",
+                    context.tp, context.epoch, partitionEpoch);
+            }
+        });
+    }
+
+    /**
+     * Schedules the unloading of a coordinator. This is called when the broker is not the
+     * leader anymore.
+     *
+     * @param tp                The topic partition of the coordinator.
+     * @param partitionEpoch    The partition epoch.
+     */
+    public void scheduleUnloadOperation(
+        TopicPartition tp,
+        int partitionEpoch
+    ) {
+        log.info("Scheduling unloading of metadata for {} with epoch {}", tp, partitionEpoch);
+
+        scheduleInternalOperation("UnloadCoordinator(tp=" + tp + ", epoch=" + partitionEpoch + ")", tp, () -> {
+            CoordinatorContext context = contextOrThrow(tp);
+            if (context.epoch < partitionEpoch) {
+                log.info("Started unloading metadata for {} with epoch {}.", tp, partitionEpoch);
+                context.transitionTo(CoordinatorState.CLOSED);
+                coordinators.remove(tp);
+                log.info("Finished unloading metadata for {} with epoch {}.", tp, partitionEpoch);
+            } else {
+                log.info("Ignored unloading metadata for {} in epoch {} since current epoch is {}.",
+                    tp, partitionEpoch, context.epoch);
+            }
+        });
+    }
+
+    /**
+     * Closes the runtime. This closes all the coordinators currently registered
+     * in the runtime.
+     *
+     * @throws Exception
+     */
+    public void close() throws Exception {
+        log.info("Closing coordinator runtime.");
+        // This close the processor, drain all the pending events and
+        // reject any new events.
+        processor.close();
```

Comment:
I am not sure. I definitely use CloseQuietly when closing the runtime component but it seems better to raise the exception further if closing the processor fails here. That is because the rest of the closing procedure does not make sense if the closing the processor has failed.